csharp/Ice/Callback/Server/SimpleWakeUpService.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) ZeroC, Inc.
+
+using EarlyRiser;
+
+namespace Server;
+
+/// <summary>SimpleWakeUpService is an Ice servant that implements Slice interface WakeUpService.</summary>
+internal class SimpleWakeUpService : WakeUpServiceDisp_
+{
+    // Implements the abstract method WakeMeUp from the WakeUpServiceDisp_ class generated by the Slice compiler.
+    public override void WakeMeUp(AlarmClockPrx? alarmClock, long timeStamp, Ice.Current current)
+    {
+        if (alarmClock is null)
+        {
+            // Should never happen, but in case it does, the Ice runtime will send an Ice.UnknownException to the
+            // client.
+            throw new ArgumentNullException(nameof(alarmClock));
+        }
+
+        var timeStampDateTime = new DateTime(timeStamp, DateTimeKind.Utc);
+
+        Console.WriteLine(
+            $"Dispatching wakeMeUp request {{ alarmClock = '{alarmClock}', timeStamp = '{timeStampDateTime.ToLocalTime()}' }}");
+
+        // Start a background task to ring the alarm clock.
+        _ = Task.Run(async () =>
+        {
+            // Wait until the specified time.
```

Comment:
Should we handle exceptions here, event if it is just to ignore them? We don't want an unobserved task exception to terminate the server process.

Reply:
I thought the default was to terminate the process.

> This event provides a mechanism to prevent exception escalation policy (which, by default, terminates the process) from triggering.

https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskscheduler.unobservedtaskexception?view=net-9.0

Reply:
See the Remarks section on that page. Or give it a try with this demo.